src/Microsoft.DotNet.Interactive.Http/HttpKernel.cs

```diff
@@ -316,8 +319,198 @@ private List<Diagnostic> GetAllDiagnostics(HttpRequestParseResult parseResult)
         if (_variables.TryGetValue(expression, out var value))
         {
             return node.CreateBindingSuccess(value);
+        } else if(MatchExpressionValue(expression, out var expressionResolution))
+        {
+            return node.CreateBindingSuccess(expressionResolution);
         }
 
         return node.CreateBindingFailure(HttpDiagnostics.CannotResolveSymbol(variableName));
     }
+
+    private bool MatchExpressionValue(string expression, out object expressionResolution)
+    {
+        expressionResolution = null;
+
+        var guidPattern = new Regex(@$"\{"$guid"}", RegexOptions.Compiled);
+        var dateTimePattern = new Regex(@$"\{"$datetime"}\s(?<type>rfc1123|iso8601|'.+'|"".+"")(?:\s(?<offset>-?\d+)\s(?<option>y|M|Q|w|d|h|m|s|ms))?", RegexOptions.Compiled);
+        var localDateTimePattern = new Regex(@$"\{"$localDatetime"}\s(?<type>rfc1123|iso8601|'.+'|"".+"")(?:\s(?<offset>-?\d+)\s(?<option>y|M|Q|w|d|h|m|s|ms))?", RegexOptions.Compiled);
+        var randomIntPattern = new Regex(@$"\{"$randomInt"}(?:\s(?<parameters>-?\d+)){{0,2}}", RegexOptions.Compiled);
+        var timestampPattern = new Regex(@$"\{"$timestamp"}(?:\s(?<offset>-?\d+)\s(?<option>y|M|Q|w|d|h|m|s|ms))?", RegexOptions.Compiled);
+
+        var guidMatches = guidPattern.Matches(expression);
+        var dateTimeMatches = dateTimePattern.Matches(expression);
+        var localDateTimeMatches = localDateTimePattern.Matches(expression);
+        var randomIntMatches = randomIntPattern.Matches(expression);
+        var timestampMatches = timestampPattern.Matches(expression);
+
+        if (guidMatches.Count > 0)
+        {
+            expressionResolution = Guid.NewGuid().ToString();
+            return true;
+        } else if(dateTimeMatches.Count > 0)
+        {
+            expressionResolution = ParseDateTime(expression, dateTimeMatches, dateTimePattern);
+            return true;
+        } else if(localDateTimeMatches.Count > 0)
+        {
+            expressionResolution = ParseDateTime(expression, localDateTimeMatches, localDateTimePattern);
+            return true;
+        } else if(randomIntMatches.Count > 0)
+        {
+            expressionResolution = ParseRandInt(expression, randomIntMatches);
+            return true;
+        }else if(timestampMatches.Count > 0)
+        {
+            expressionResolution = ParseTimestamp(expression, timestampMatches);
+            return true;
+        }
+        return false;
+    }
+
+    protected string ParseTimestamp(string expression, MatchCollection matches)
+    {
+        var text = expression;
+        var currentDateTimeOffset = DateTimeOffset.UtcNow;
+        for (int i = matches.Count - 1; i >= 0; i--)
```

Comment:
Why do we expect more than one match? Each embedded expression i.e. `{{ }}` only contain at most one expression, isn't it?